core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -10171,17 +10164,26 @@ matched clause of merge). If the upsert caused a row to be updated in the
 base table then the old version of the row will have to be deleted from 
 indexes, and a new version inserted. Upsert is being transformed to merge
 so that we can delete the old version of an updated row from the index.
+
+Upsert is also converted into merge when there are omitted cols with default values and 
+TRAF_UPSERT_WITH_INSERT_DEFAULT_SEMANTICS is set to  OFF in case of aligned format table or 
+omitted current timestamp cols in case of non-aligned row format
```

Comment:
Thanks, that's good news. So, my proposal would be to stop removing those columns in the base table insert (unless an override CQD is set).

Yes, if we always set the timestamp that would overwrite an existing row's timestamp, and I believe that is the right thing to do. I'll send out an email to the user list to ask for others' opinions.